src/arch/xtensa/smp/idc.c

```diff
@@ -34,7 +34,7 @@ void idc_enable_interrupts(int target_core, int source_core)
 {
 	idc_write(IPC_IDCCTL, target_core,
 		  IPC_IDCCTL_IDCTBIE(source_core));
-	platform_interrupt_unmask(PLATFORM_IDC_INTERRUPT(target_core));
+	interrupt_cascade_unmask(PLATFORM_IDC_INTERRUPT(target_core));
```

Comment:
Can we not just call interrupt_unmask(), the clients dont care about the IRQ implementation. ditto for others, we can remove cascade from external API names.

Reply:
Same reasoning basically - I wanted to keep those APIs apart while performing changes. To avoid breaking anything. Once we have a reviewed verified working state, we can merge APIs together.